sdk/src/main/scala/com/horizen/account/state/StateTransition.scala

```diff
@@ -105,10 +108,20 @@ class StateTransition(view: AccountStateView, messageProcessors: Seq[MessageProc
   private def refundGas(msg: Message, gas: GasPool): Unit = {
     val usedGas = msg.getGasLimit.subtract(gas.getGas)
     // cap gas refund to a quotient of the used gas
-    gas.addGas(view.getRefund.min(usedGas.divide(GasUtil.RefundQuotientEIP3529)))
+    val refGas = view.getRefund.min(usedGas.divide(GasUtil.RefundQuotientEIP3529))
+    gas.addGas(refGas)
     // return funds for remaining gas, exchanged at the original rate.
     val remaining = gas.getGas.multiply(msg.getGasPrice)
-    view.addBalance(msg.getFrom.address(), remaining)
+    val sender = msg.getFrom.address()
+    log.debug(s"gas used: $usedGas, remaining gas: ${gas.getGas}, refGas: $refGas, gasPrice: ${msg.getGasPrice}")
```

Comment:
`usedGas` will be outdated here because we added some gas back to the pool in line 112.

Reply:
Agree

Reply:
this one is still open

Reply:
Ok, done